plugins/generic/externalFeed/ExternalFeedPlugin.inc.php

```diff
@@ -1,375 +1,329 @@
+			
+			$externalFeedDao = new ExternalFeedDAO($this->getName());
+			$returner =& DAORegistry::registerDAO('ExternalFeedDAO', $externalFeedDao);
+			
+			HookRegistry::register('Templates::Management::Settings::website', array($this, 'callbackShowWebsiteSettingsTabs'));
+			HookRegistry::register('LoadComponentHandler', array($this, 'setupGridHandler'));
+			
+			$templateMgr =& TemplateManager::getManager();
+			$templateMgr->addStyleSheet('externalFeed', '/' . $this->getStyleSheetFile());
+
+			// Journal home page display
+			HookRegistry::register('TemplateManager::display', array($this, 'displayHomepage'));
+
+			$this->_registerTemplateResource();
+		}
+		return $success;
+	}
+
+	/**
+	 * Get the display name of this plugin
+	 * 
+	 * @return string
+	 */
+	public function getDisplayName() {
+		return __('plugins.generic.externalFeed.displayName');
+	}
+
+	public function getDescription() {
+		return __('plugins.generic.externalFeed.description');
+	}
+
+	/**
+	 * Get the filename of the ADODB schema for this plugin.
+	 */
+	public function getInstallSchemaFile() {
+		return $this->getPluginPath() . '/' . 'schema.xml';
+	}
+
+	/**
+	 * Get plugin CSS URL
+	 *
+	 * @return string Public plugin CSS URL
+	 */
+	public function getCssUrl() {
+		return $this->getPluginPath() . '/css/';
+	}
+
+	/**
+	 * Get the filename of the default CSS stylesheet for this plugin.
+	 * @return string 
+	 */
+	public function getDefaultStyleSheetFile() {
+		return $this->getCssUrl() . 'externalFeed.css';
+	}
+
+	/**
+	 * Override the builtin to get the correct template path.
+	 * @return string
+	 */
+	public function getTemplatePath($inCore = false)
+	{
+		return $this->getTemplateResourceName() . ':templates/';
+	}
+
+	/**
+	 * @see Plugin::getActions()
+	 */
+	function getActions($request, $verb) {
+		import('lib.pkp.classes.linkAction.request.AjaxModal');
+		import('lib.pkp.classes.linkAction.request.RedirectAction');
+
+		$router = $request->getRouter();
+		$dispatcher = $request->getDispatcher();
+
+		return array_merge(
+				$this->getEnabled()?array(
+						new LinkAction(
+								'settings',
+								new AjaxModal(
+										$router->url($request, null, null, 'manage', null, array('verb' => 'settings', 'plugin' => $this->getName(), 'category' => 'generic')),
+										$this->getDisplayName()
+								),
+								__('manager.plugins.settings'),
+								null
+						),
+				):array(),
+				$this->getEnabled()?array(
+						new LinkAction(
+								'feeds',
+								new RedirectAction($dispatcher->url(
+									$request, ROUTE_PAGE,
+									null, 'management', 'settings', 'website',
+									array('uid' => uniqid()), // Force reload
+									'externalFeeds'
+								)),
+								__('plugins.generic.externalFeed.manager.feeds'),
+								null
+						),
+				):array(),
+				parent::getActions($request, $verb)
+		);
+	}
+
+	/**
+	 * @see Plugin::manage()
+	 */
+	function manage($args, $request) {
+
+		$journal = $request->getJournal();

```

Comment:
Still needs review for OJS-specific language; see https://github.com/pkp/ojs/pull/1115#discussion_r161036124